googlecloudstorage_blueprints/move_file.py

```diff
@@ -0,0 +1,230 @@
+    """
+    Fetched all the files in the bucket which are returned in a list as
+    Google Blob objects
+    """
+    return list(bucket.list_blobs(prefix=prefix))
+
+
+def download_google_cloud_storage_file(blob, destination_file_name=None):
+    """
+    Download a selected file from Google Cloud Storage to local storage in
+    the current working directory.
+    """
+    local_path = os.path.normpath(f'{os.getcwd()}/{destination_file_name}')
+
+    blob.download_to_filename(local_path)
+
+    print(f'{blob.bucket.name}/{blob.name} successfully downloaded to {local_path}')
+
+    return
+
+
+def get_gclient(args):
+    """
+    Attempts to create the Google Cloud Storage Client with the associated
+    environment variables
+    """
+    try:
+        gclient = storage.Client()
+    except Exception:
+        print(f'Error accessing Google Cloud Storage with service account ',
+              f'{args.gcp_application_credentials}')
+        sys.exit(ec.EXIT_CODE_INVALID_CREDENTIALS)
+
+    return gclient
+
+
+def get_bucket(*,
+               gclient,
+               bucket_name):
+    """
+    Fetches and returns the bucket from Google Cloud Storage
+    """
+    try:
+        bucket = gclient.get_bucket(bucket_name)
+    except Exception as e:
+        print(f'Bucket {bucket_name} does not exist\n {e}')
+        sys.exit(ec.EXIT_CODE_INVALID_BUCKET)
+
+    return bucket
+
+
+def get_storage_blob(bucket, source_folder_name, source_file_name):
+    """
+    Fetches and returns the single source file blob from the buck on
+    Google Cloud Storage
+    """
+    source_path = source_file_name
+    if source_folder_name != '':
+        source_path = f'{source_folder_name}/{source_file_name}'
+    blob = bucket.get_blob(source_path)
+    try:
+        blob.exists()
+        return blob
+    except Exception as e:
+        print(f'File {source_path} does not exist')
+        sys.exit(ec.EXIT_CODE_FILE_NOT_FOUND)
+
+
+def move_google_cloud_storage_file(source_bucket, source_blob_path, 
+                                destination_bucket, destination_blob_path):
+    """
+    Moves blobs between directories or buckets. First copies the  
+    blob to destination then deletes the source blob from the old location.
+    """
+    # get source blob
+    source_blob = source_bucket.blob(source_blob_path)
+
+    # copy to destination
+    dest_blob = source_bucket.copy_blob(
+        source_blob, destination_bucket, destination_blob_path)
+    # delete in old destination
+    source_blob.delete()
+    
+    print(f'File moved from {source_blob} to {dest_blob}')
+
+
+def gcp_find_matching_files(file_blobs, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for blob in file_blobs:
+        if re.search(file_name_re, blob.name):
+            matching_file_names.append(blob)
+
+    return matching_file_names
+
```

Comment:
For sustainability, I'd prefer to remove reference to this function and use files.find_all_file_matches() from shipyard_utils. It does exactly the same thing, with the addition of printing the number of matches and their names (which means some of your current prints might need to be rearranged).

Want to centralize this so if the logic ever has to be updated, it gets updated in every Blueprint.